### PR TITLE
feat: implementar plano de tratamento do paciente no frontend

### LINKED ISSUE #59: Implementar Plano de Tratamento do paciente no frontend
## Descrição do problema ou necessidade

A partir da atividade de backend [Implementar Plano de Tratamento do paciente #51](https://github.com/FabioCarlesso/carlessopilatesapi/issues/51), o frontend precisa permitir que o fisioterapeuta cadastre, consulte, atualize e acompanhe os planos de tratamento vinculados a um paciente.

Após a anamnese e a avaliação fisioterapêutica, o sistema deve oferecer uma interface clara para registrar a estratégia terapêutica do paciente, incluindo objetivos, frequência semanal, condutas propostas, exercícios indicados/contraindicados, equipamentos previstos, observações clínicas e status do plano.

## Objetivo da atividade

Implementar no frontend o módulo/tela de Plano de Tratamento do paciente, integrando com os endpoints disponibilizados pelo backend e permitindo o gerenciamento do histórico de planos por paciente.

## Escopo da implementação

- Criar tela ou seção de **Plano de Tratamento** dentro do contexto/detalhe do paciente.
- Criar formulário para cadastro e edição do plano de tratamento.
- Exibir histórico/listagem de planos de tratamento do paciente.
- Permitir visualizar os detalhes de um plano específico.
- Permitir identificar visualmente o plano ativo do paciente.
- Permitir encerrar, suspender ou inativar um plano de tratamento sem exclusão física.
- Integrar com os endpoints do backend para:
  - Criar plano de tratamento para um paciente.
  - Buscar plano por ID.
  - Listar planos de um paciente.
  - Atualizar plano de tratamento.
  - Encerrar/inativar plano de tratamento.
- Implementar campos mínimos no formulário:
  - Data de início.
  - Data prevista de término, se houver.
  - Objetivos terapêuticos.
  - Frequência semanal.
  - Condutas propostas.
  - Exercícios indicados.
  - Exercícios ou movimentos contraindicados.
  - Equipamentos previstos.
  - Observações clínicas.
  - Status do plano: ativo, encerrado ou suspenso.
- Implementar validações de campos obrigatórios conforme contrato do backend.
- Tratar mensagens de sucesso e erro nas operações principais.
- Manter padrão visual e estrutural já utilizado no projeto Angular.

## Critérios de aceite

- Deve existir uma área de Plano de Tratamento acessível a partir do detalhe do paciente.
- Deve ser possível criar um plano de tratamento para um paciente existente.
- Deve ser possível editar um plano de tratamento existente.
- Deve ser possível listar o histórico de planos de tratamento de um paciente.
- Deve ser possível visualizar os dados completos de um plano de tratamento.
- Deve ser possível identificar qual plano está ativo.
- Deve ser possível encerrar, suspender ou inativar um plano sem removê-lo da listagem.
- O formulário deve validar os campos obrigatórios antes do envio.
- O frontend deve tratar erro ao tentar operar sobre paciente inexistente ou plano inexistente, caso o backend retorne erro.
- O frontend deve exibir feedback visual para criação, atualização e inativação do plano.
- A implementação deve seguir o padrão de organização atual do projeto, incluindo componentes, services, models/interfaces e rotas.

## Sugestão de validação/testes

- Validar criação de plano com dados obrigatórios preenchidos.
- Validar bloqueio do envio quando campos obrigatórios estiverem vazios.
- Validar carregamento da listagem de planos por paciente.
- Validar exibição correta do plano ativo.
- Validar edição de um plano existente.
- Validar encerramento/inativação de um plano.
- Validar tratamento de erro quando o backend retornar paciente não encontrado.
- Validar tratamento de erro quando o backend retornar plano não encontrado.
- Criar testes unitários para o service responsável pela integração com a API.
- Criar testes unitários para validações principais do formulário.
- Quando aplicável, criar teste de componente para garantir renderização da listagem e do formulário.

## Observações técnicas

- A implementação depende da disponibilidade/contrato final dos endpoints definidos na issue de backend #51.
- Recomenda-se criar uma interface/model para `PlanoTratamento` e DTOs de request/response compatíveis com a API.
- Avaliar se o status será controlado por enum no frontend: `ATIVO`, `ENCERRADO`, `SUSPENSO` ou conforme retorno real do backend.
- Os campos textuais podem iniciar como texto livre, alinhado à primeira versão do backend.
- Caso o backend defina a regra de apenas um plano ativo por paciente, o frontend deve refletir essa regra na interface e nas mensagens ao usuário.
- Futuramente, sessões/aulas poderão ser vinculadas ao plano de tratamento ativo.

## Labels sugeridas

- `enhancement`
- `frontend`
- `angular`
- `plano-tratamento`
- `prontuario`

### PR DESCRIPTION
**Pull request by FabioCarlesso**
## Resumo

Implementação do módulo de Plano de Tratamento do paciente no frontend Angular, integrando com os endpoints do backend.

## Alterações

- **`src/app/core/models/plano-tratamento.ts`** — Model com interfaces `PlanoTratamentoRequestDTO`, `PlanoTratamentoUpdateDTO`, `PlanoTratamentoResponseDTO` e tipo `PlanoTratamentoStatus` (`ATIVO` | `ENCERRADO` | `SUSPENSO`)
- **`src/app/core/services/plano-tratamento.service.ts`** — Service com os métodos: `listarPorPaciente`, `buscar`, `criar`, `atualizar`, `encerrar`, `suspender`
- **`paciente-plano-tratamento-list/`** — Componente de listagem de planos por paciente com suporte a encerrar/suspender com confirmação
- **`paciente-plano-tratamento-form/`** — Formulário de criação e edição com todos os campos obrigatórios e opcionais da issue
- **`app.routes.ts`** — Novas rotas: `/pacientes/:pacienteId/plano-tratamento`, `/pacientes/:pacienteId/plano-tratamento/novo` e `/pacientes/:pacienteId/plano-tratamento/:id/editar`
- **`paciente-detail.component.html`** — Link "Plano de Tratamento" adicionado na seção de Gestão do detalhe do paciente

## Motivo

Resolver a issue #59: permitir que o fisioterapeuta cadastre, consulte, atualize e acompanhe os planos de tratamento vinculados a um paciente após anamnese e avaliação fisioterapêutica.

## Testes executados

- `ng test --watch=false --browsers=ChromeHeadless` → **334 testes passando, 0 falhas**
- `ng build` → Build concluído sem erros

### Cobertura de testes criados

- **`plano-tratamento.service.spec.ts`** — 6 testes: GET lista, GET por id, POST, PUT, PATCH encerrar, PATCH suspender
- **`paciente-plano-tratamento-list.component.spec.ts`** — 13 testes: carregamento, estado vazio, múltiplos planos, erros, confirmar/cancelar/executar ações, labels
- **`paciente-plano-tratamento-form.component.spec.ts`** — 11 testes: modo criação (campos obrigatórios, validações, whitespace, range), modo edição (pre-fill, atualizar, erro), parâmetro inválido

## Referência

Closes #59

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/app/pages/pacientes/paciente-plano-tratamento-form/paciente-plano-tratamento-form.component.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    const plano$ = this.planoId !== null
+      ? this.planoTratamentoService.buscar(this.planoId)
+      : of(null);
+
+    forkJoin({ paciente: paciente$, plano: plano$ })
+      .pipe(takeUntilDestroyed(this.destroyRef))
+      .subscribe({
+        next: ({ paciente, plano }) => {
+          this.paciente = paciente;
+          this.plano = plano;
```
Comment (FabioCarlesso) [inline src/app/pages/pacientes/paciente-plano-tratamento-form/paciente-plano-tratamento-form.component.ts]: Problem: edit mode loads the plan by raw `planoId` and then trusts it without checking that `plano.pacienteId` matches the `pacienteId` from the route.

Why it matters: a URL like `/pacientes/10/plano-tratamento/123/editar` can show patient 10 in the header while editing plan 123 even if that plan belongs to another patient. If the backend permits the lookup, this can expose or mutate clinical data under the wrong patient context.

Suggested improvement: after loading, reject mismatches before patching the form, e.g. compare `plano.pacienteId !== this.pacienteId`, show an error and/or navigate back. Better still, use a backend endpoint scoped by patient when available.

Severity: High

File: src/app/pages/pacientes/paciente-plano-tratamento-form/paciente-plano-tratamento-form.component.ts [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+      exerciciosContraindicados: opt(valor.exerciciosContraindicados),
+      equipamentosPrevistos: opt(valor.equipamentosPrevistos),
+      observacoesClinicas: opt(valor.observacoesClinicas),
+      status: valor.status as PlanoTratamentoStatus
+    };
+
+    const planoAtual = this.plano;
+    const criando = planoAtual === null;
+    const obs = criando
+      ? this.planoTratamentoService.criar({ pacienteId: this.pacienteId, ...dto })
```
Comment (FabioCarlesso) [inline src/app/pages/pacientes/paciente-plano-tratamento-form/paciente-plano-tratamento-form.component.ts]: Problem: the create request sends the shared `dto`, which includes `status`, even though `PlanoTratamentoRequestDTO` does not define `status`.

Why it matters: this diverges from the typed API contract and can either break against a strict backend (`400` for an unknown field) or allow the client to create a plan directly as `ENCERRADO`/`SUSPENSO` if the backend accepts it. Creation should usually let the backend choose the initial state.

Suggested improvement: build separate create/update payloads or destructure `status` out for create: `const { status, ...createDto } = dto; this.planoTratamentoService.criar({ pacienteId: this.pacienteId, ...createDto })`. Add a service/component test asserting `status` is not sent on POST.

Severity: Medium

File: src/app/pages/pacientes/paciente-plano-tratamento-list/paciente-plano-tratamento-list.component.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      ? this.planoTratamentoService.encerrar(id)
+      : this.planoTratamentoService.suspender(id);
+
+    obs.subscribe({
+      next: () => {
+        this.sucesso = acao === 'encerrar'
+          ? 'Plano de tratamento encerrado com sucesso.'
+          : 'Plano de tratamento suspenso com sucesso.';
+        this.carregarPlanos();
+        setTimeout(() => { this.sucesso = null; }, 4000);
```
Comment (FabioCarlesso) [inline src/app/pages/pacientes/paciente-plano-tratamento-list/paciente-plano-tratamento-list.component.ts]: Problem: this component creates subscriptions and a timeout but has no teardown path.

Why it matters: if the user navigates away while a request or the success timer is pending, the callback can still mutate a destroyed component. It is a small leak here, but it is inconsistent with the neighboring patient clinical components, which use `DestroyRef`/`takeUntilDestroyed` and clear success timers.

Suggested improvement: inject `DestroyRef`, pipe the three service calls through `takeUntilDestroyed(this.destroyRef)`, store the success timer id, and clear it in `ngOnDestroy`.

Severity: Low

File: src/app/pages/pacientes/paciente-plano-tratamento-list/paciente-plano-tratamento-list.component.html [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+          <span class="label">Observações clínicas</span>
+          <span>{{ plano.observacoesClinicas }}</span>
+        </div>
+      </div>
+    </div>
+  </div>
+</div>
+
+<div class="dialog-overlay" *ngIf="confirmarAcaoId !== null">
+  <div class="dialog">
```
Comment (FabioCarlesso) [inline src/app/pages/pacientes/paciente-plano-tratamento-list/paciente-plano-tratamento-list.component.html]: Problem: the custom confirmation dialog has no dialog semantics or focus handling.

Why it matters: screen readers will not get a modal announcement, keyboard users are not moved into the dialog, and focus can remain behind the overlay. That makes destructive actions like `Encerrar`/`Suspender` harder to operate safely.

Suggested improvement: add `role="dialog"`, `aria-modal="true"`, an accessible label/heading, and focus management on open/close, or use a shared accessible dialog component if the project has one.

Severity: Low

**Review by FabioCarlesso — COMMENTED** [review]
I reviewed the PR across documentation, bugs/edge cases, security, performance, Angular practices, API integration, UI/UX, tests, and code quality. Tests/build passed locally (`334` Karma tests and `npm run build`). I left inline comments for the issues I found.

Areas that look good: route ordering is covered by tests; the new service has HTTP method/path tests; invalid numeric route params are handled; templates use Angular interpolation rather than unsafe HTML; no secrets or obvious token-handling regressions were introduced; bundle impact is lazy-loaded and small.

**Comment by FabioCarlesso** [comment]
Problem: the implementation adds a new treatment-plan flow, but the frontend documentation was not updated. `README.md` still omits the new patient treatment-plan folders from the structure section, the test file list, the implemented modules table, and the routes table. `docs/context.md` still lists the current state through Avaliação Fisioterapêutica only and its backend contract table does not include `GET/POST/PUT/PATCH /planos-tratamento`.

Why it matters: this repo uses `README.md` and `docs/context.md` as the source of onboarding/context for implemented frontend modules. After this PR, a developer following the docs will not discover the new routes, service contract, status actions, or tests.

Suggested improvement: update the README and context docs with the new routes (`/pacientes/:pacienteId/plano-tratamento`, `/novo`, `/:id/editar`), service endpoints, DTO/status behavior, new component/test paths, and the link from patient detail.

Severity: Medium

**Comment by FabioCarlesso** [comment]
Correções aplicadas no commit `d73bf31` e push feito para `issue-59-plano-tratamento-paciente`.

Resumo dos ajustes:
- Documentação atualizada em `README.md`, `docs/context.md` e `docs/documentacao.md` com rotas, endpoints, estrutura e testes do Plano de Tratamento.
- Formulário de edição agora valida se o plano carregado pertence ao `pacienteId` da rota antes de exibir/preencher dados.
- Payload de criação não envia mais `status`; status fica apenas no DTO de atualização.
- Listagem agora usa `takeUntilDestroyed`, limpa timer no destroy e adiciona semântica/foco ao modal de confirmação.
- Testes adicionados para criação sem `status`, plano de outro paciente e `id` inválido.

Validação local:
- `npm test -- --watch=false --browsers=ChromeHeadless` → 337 testes passando
- `npm run build` → build concluído
